lib/controllers/project.js

```diff
@@ -693,6 +724,80 @@ Project.prototype.packageProject = function(dir, includeModules, cb) {
 
 //-----------------------------------------------------------------------------
 /**
+  * Detects the type of project being deployed.
+  * @param {string} path The path being deployed.
+  * @param {function} callback
+  */
+//-----------------------------------------------------------------------------
+Project.prototype.detectProjectType = function (path, callback) {
+  var projectController = this;
+
+  async.waterfall([
+    function (next) {
+      projectController.isMeteor(path, function (err, result) {
+        if (err) return callback(err);
+        if (result) return next('METEOR', true);
+        next();
+      });
+    }
+  ], function (err) {
+    var result;
+
+    if (typeof err  !== 'object' && Project.types[err] !== 'undefined') {
+      result = err;
+      err = null;
+    }
+
+    callback(err, Project.types[result] || Project.types.NODEJS); // Default type to Node.js
+  });
+};
+
+//-----------------------------------------------------------------------------
+/**
+  * Identifies if the source code version is supported by build/app servers.
+  * @param {String} srcVersion The version of the deploying application.
+  * @param {String} runtime The runtime of project being deployed.
+  * @param {Boolean} Returns boolean
+  */
+//-----------------------------------------------------------------------------
+Project.prototype.isSupported = function (srcVersion, runtime) {
+  if (runtime !== 'Meteor') return true;
+
+  srcVersion = srcVersion.split('.')
+    .slice(0, 3)
+    .join('.');
+
+  return semver.gte(srcVersion, Project.minVersions[runtime]);
+};
+
+//-----------------------------------------------------------------------------
+/**
+  * Detects whether or not the project being deployed is a Meteor application.
+  * Looks for a .meteor folder that contains a packages file.
+  * @param {String} dir The path being deployed.
+  * @param {Function} callback Invoked with (err, true|false)
+  */
+//-----------------------------------------------------------------------------
+Project.prototype.isMeteor = function (dir, callback) {
+  async.waterfall([
+    fs.readdir.bind(null, dir),
+    function (files, next) {
+      if (files.indexOf('.meteor') !== -1) {
+        fs.readdir(path.join(dir, '.meteor'), next);
+      } else next('ok', false);
+    },
+    function (files, next) {
+      if (files.indexOf('packages') !== -1) next(null, true);
+      else next(null, false);
+    }
+  ], function (err, result) {
+    if (err === 'ok') err = null;
+    callback(err, result);
```

Comment:
passing null means the `waterfall` calls the next function, this is an escape hatch